app/views/step_by_step_pages/show/_actions.erb

```diff
@@ -33,7 +33,13 @@
         class: %w(govuk-link govuk-link--no-visited-state app-link--right)
       %>
     <% elsif @step_by_step_page.has_draft? %>
-      <% if @step_by_step_page.status.draft? &&
+      <% if @step_by_step_page.status.submitted_for_2i? && can_review?(@step_by_step_page, current_user) %>
```

Comment:
I'd move `submitted_for_2i` into `can_review` so this template is literally just `<% if can_review?(...) %>`

Reply:
Moved as suggested. It's still a bit unclear to me what do we keep in the `helpers/step_nav_action_helper.rb` and what in `model/step_by_step_page.rb`.